common/src/graph_collision.rs

```diff
@@ -0,0 +1,559 @@
+use std::collections::VecDeque;
+
+use fxhash::FxHashSet;
+
+use crate::{
+    chunk_collision::chunk_sphere_cast,
+    dodeca::{self, Vertex},
+    math,
+    node::{Chunk, ChunkId, ChunkLayout, DualGraph},
+    proto::Position,
+};
+
+/// Performs sphere casting (swept collision query) against the voxels in the `DualGraph`
+///
+/// The `ray` parameter and any resulting hit normals are given in the local coordinate system of `position`.
+///
+/// The `tanh_distance` is the hyperbolic tangent of the cast_distance, or the distance along the ray to check for hits.
+///
+/// This function may return a `SphereCastError` if not enough chunks are generated, even if the ray never reaches an
+/// ungenerated chunk. To prevent these errors, make sure that the distance between the ray's start point and the closest
+/// ungenerated chunk's center is less than `cast_distance + collider_radius + dodeca::BOUNDING_SPHERE_RADIUS`
```

Comment:
Should this say "greater than", and with respect to the node center, rather than chunk center?

Reply:
Good catch. I was picturing as a ray needing to be short enough, hence the "less than", but I did get it backwards.